pkg/context/context.go

```diff
@@ -109,3 +157,48 @@ func checkEC2Connectivity(ctx context.Context, api *ec2.EC2) error {
 	}
 	return err
 }
+
+func ResolveClusterEndpoint(ctx context.Context, eksAPI eksiface.EKSAPI) (string, error) {
```

Comment:
I don't think this needs to be public, just put the test that is validating that this works in this package